GDJS/Runtime/spriteruntimeobject.ts

```diff
@@ -475,40 +475,29 @@ namespace gdjs {
       const direction = this._animations[this._currentAnimation].directions[
         this._currentDirection
       ];
-      const oldFrame = this._currentFrame;
-      const animationDuration =
-        direction.frames.length * direction.timeBetweenFrames;
+      const animationDuration = this.getAnimationDuration();
       if (
         !this._animationPaused &&
         (direction.loop || this._animationElapsedTime !== animationDuration) &&
         direction.timeBetweenFrames
       ) {
-        const elapsedTime = this.getElapsedTime() / 1000;
-        this._animationElapsedTime += elapsedTime * this._animationSpeedScale;
-        if (direction.loop) {
-          this._animationElapsedTime = gdjs.evtTools.common.mod(
-            this._animationElapsedTime,
-            direction.frames.length * direction.timeBetweenFrames
-          );
-        } else {
-          this._animationElapsedTime = gdjs.evtTools.common.clamp(
-            this._animationElapsedTime,
-            0,
-            animationDuration
-          );
-        }
-        this._currentFrame = Math.min(
-          Math.floor(this._animationElapsedTime / direction.timeBetweenFrames),
-          direction.frames.length - 1
+        const animationElapsedTime =
+          this._animationElapsedTime +
+          (this.getElapsedTime() / 1000) * this._animationSpeedScale;
+        this.setAnimationElapsedTime(
+          direction.loop
+            ? gdjs.evtTools.common.mod(animationElapsedTime, animationDuration)
+            : gdjs.evtTools.common.clamp(
```

Comment:
In theory this clamp is not needed as done by setAnimationElapsedTime?

Reply:
Yes, it's for clarity.